bpf/lib/lb.h

```diff
@@ -188,6 +210,28 @@ bool lb6_svc_is_hostport(const struct lb6_service *svc __maybe_unused)
 #endif /* ENABLE_HOSTPORT */
 }
 
+static __always_inline
+bool lb4_svc_is_lb_with_src_range_check(const struct lb4_service *svc __maybe_unused)
+{
+#ifdef ENABLE_LB_SRC_RANGE_CHECK
+	return (svc->flags & (SVC_FLAG_LOADBALANCER | SVC_FLAG_SOURCE_RANGE)) ==
+		(SVC_FLAG_LOADBALANCER | SVC_FLAG_SOURCE_RANGE);
```

Comment:
I actually think the implementation here should not mention anything that is tied to LB svc type since it really can be decoupled from it given there are no LB type specific dependencies. Similarly, I would change `--enable-source-range-check` flag for the agent (+ Helm), meaning I would also remove any LB mention from the agent flag. The documentation is the only place where it should provide a small howto for guiding how to use loadBalancerSourceRanges with the agent. But in future this functionality could also be reused for other service types if we ever need to (even if just Cilium specific annotation).

Reply:
I thought about having it generic (i.e. not tied to a svc type). However, currently the LB src range check can only be performed for a `LoadBalancer` service, as for other service types k8s rejects the svc object with the field being set. E.g.:
```
The Service "nginx-service" is invalid: spec.LoadBalancerSourceRanges: Forbidden: may only be used when `type` is 'LoadBalancer'
```
Therefore, I decided to stick to the k8s service spec, and to make it LB specific.

Reply:
Right, but that is the /current/ K8s state which can change/extend depending on use-case. We could also add our own Cilium annotation if needed in future for other services. I think this one makes sense also for the case of services w/ external IP for example. So I would really make this a generic facility (since it really is generic and it would simplify the code a bit), and regard the K8s loadBalancerSourceRanges as /one/ user of it.